src/Client/Updater.php

```diff
@@ -546,9 +546,20 @@ public function download(string $target): PromiseInterface
         $length = $targetsMetaData->getLength($target) ?? static::MAXIMUM_DOWNLOAD_BYTES;
 
         $verify = function (StreamInterface $stream) use ($target, $hashes) {
-            $content = $stream->getContents();
+            // If the stream has a URI that refers to a file, use hash_file() to
+            // verify it. Otherwise, read the entire stream as a string and use
+            // hash() to verify it.
+            $uri = $stream->getMetadata('uri');
+            if ($uri && file_exists($uri)) {
+                $content = $uri;
+                $verifier = 'hash_file';
+            } else {
+                $content = $stream->getContents();
+                $verifier = 'hash';
+            }
+
             foreach ($hashes as $algo => $hash) {
-                if ($hash !== hash($algo, $content)) {
+                if ($hash !== $verifier($algo, $content)) {
```

Comment:
This seems weird to use a variable as the function call if we don't have to.

Also is `$content` really content if it was set to `$uri`. Why not something like this

           $uri = $stream->getMetadata('uri');
            if (!$uri || !file_exists($uri)) {
                $content = $stream->getContents();
            }

            foreach ($hashes as $algo => $hash) {
                $streamHash = isset($content) ? hash($algo, $content) : hash_file($algo, $uri);
                if ($hash !== $streamHash) {
                    throw new InvalidHashException("Invalid $algo hash for $target");
                }
            }

Reply:
Fixed this by moving the existing logic into the loop and removing the function name variable.